tiatoolbox/models/segmentation/semantic_segmentor.py

```diff
@@ -0,0 +1,981 @@
+
+        Args:
+            imgs (list, ndarray): List of inputs to process. When using `patch`
+                mode, the input must be either a list of images, a list of image file
+                paths or a numpy array of an image list. When using `tile` or `wsi`
+                mode, the input must be a list of file paths.
+            masks (list): List of masks. Only utilised when processing image tiles
+                and whole-slide images. Patches are only processed if they are witin a
+                masked area. If not provided, then a tissue mask will be automatically
+                generated for whole-slide images or the entire image is processed for
+                image tiles.
+            mode (str): Type of input to process. Choose from either `tile` or `wsi`.
+            ioconfig (object): object that define information about input and ouput
+                placement of patches. When provided, `patch_input_shape`,
+                `patch_output_shape`, `stride_shape`, `resolution`, and `units`
+                arguments are ignore. Otherwise, those arguments will be internally
+                converted to an ioconfig object.
+            on_gpu (bool): whether to run model on the GPU.
+            patch_input_shape (tuple): Size of patches input to the model. The value
+                are at requested read resolution and must be positive.
+            patch_output_shape (tuple): Size of patches output by the model. The value
+                are at requested read resolution and must be positive.
+            stride_shape (tuple): Stride using during tile and WSI processing. The value
+                are at requested read resolution and must be positive.
+                If not provided, `stride_shape=patch_input_shape`.
+            resolution (float): Resolution used for reading the image.
+            units (str): Units of resolution used for reading the image. Choose from
+                either `level`, `power` or `mpp`.
+            save_dir (str): Output directory when processing multiple tiles and
+                whole-slide images. By default, it is folder `output` where the
+                running script is invoked.
+            crash_on_exception (bool): If `True`, the running loop will crash
+                if there is any error during processing a WSI. Otherwise, the loop
+                will move on to the next wsi for processing.
+
+        Returns:
+            output (list): A list of tuple(input_path, save_path) where
+                `input_path` is the path of the input wsi while `save_path`
+                corresponds to the output predictions.
+
+        Examples:
+            >>> # Sample output of a network
+            >>> wsis = ['A/wsi1.svs', 'B/wsi2.svs']
+            >>> predictor = SemanticSegmentor(model='unet')
+            >>> output = predictor.predict(wsis, mode='wsi')
+            >>> output.keys()
+            [('A/wsi.svs', 'output/0.raw') , ('B/wsi.svs', 'output/1.raw')]
```

Comment:
add `...`. Please see https://github.com/TIA-Lab/tiatoolbox/blob/develop/tiatoolbox/wsicore/save_tiles.py

Reply:
This is assumed output. I follow here
https://github.com/TIA-Lab/tiatoolbox/blob/0a26f42ed1b26149a1a8a1cb60f0f27cc14a23d2/tiatoolbox/wsicore/wsireader.py#L135